sos/report/plugins/openstack_octavia.py

```diff
@@ -63,6 +69,15 @@ def setup(self):
                 "openstack loadbalancer show %s" % (loadbalancer)
             )
 
+        # get details from each amphora
+        cmd = "openstack loadbalancer amphora list -f value -c id"
+        amphorae = self.exec_cmd(cmd)['output']
+        for amphora in amphorae.splitlines():
+            amphora = amphora.split()[0]
+            self.add_cmd_output(
+                "openstack loadbalancer amphora show %s" % (amphora)
+            )
+
```

Comment:
This gets repeated a number of times. Let's consolidate this and make it easier to add more resources in the future. We could do something like the `kubernetes` and `openshift` plugins here, E.G.:


```python
resources = [
    'amphora',
    'flavor',
    'flavorprofile'
   ... etc ...
]

for res in resources:
    cmd = "openstack loadbalancer %s list -f value -c id" % res
    ret = self.exec_cmd(cmd)
    if ret['status'] == 0:
        for ent in ret['output'].splitlines():
            ent = ent.split()[0]
            self.add_cmd_output(
                "openstack loadbalancer %s show %s" % (res, ent)
            )
```

For the `add_cmd_output()`, you could also pass `subdir=res` to get these outputs sorted by resource in `sos_commands/openstack_octavia/$res`

Reply:
Agreed. I've addressed this comment. I also fixed one other issue I found with "provider capability list".